### PR TITLE
Define custom components module & pipeline base dir globally

### PR DESCRIPTION
**Pull request by disrupted**
Depends on #347 
for added convenience, ability to set pipeline base dir and components module globally for the project. therefore it won't be necessary to specify these with each invocation
will be used for automatic schema generation (editor integration)

### CODE REVIEW

File: docs/docs/resources/variables/config_env_vars.env
Comment (disrupted) [inline docs/docs/resources/variables/config_env_vars.env:1]: Depends on #384

**Comment by disrupted** [comment]
should we remove those two options from the CLI now that they're global settings?

File: kpops/cli/main.py [hunk 1]
```diff
@@ -28,6 +28,8 @@
 
     from kpops.components.base_components import PipelineComponent
 
+ENV_PREFIX = KpopsConfig.model_config.get("env_prefix")
```
Comment (sujuka99) [inline kpops/cli/main.py:31]: This is already [defined in the config](https://github.com/bakdata/kpops/blob/v3/kpops/config.py#L17), you could import the constant from there

File: kpops/cli/main.py [hunk 2]
```diff
@@ -377,7 +423,7 @@ def reset(
         component.reset(dry_run)
 
 
-@app.command(help="Clean pipeline steps")  # pyright: ignore[reportGeneralTypeIssues] https://github.com/rec/dtyper/issues/8
+@app.command(help="Clean pipeline steps")  # pyright: ignore[reportGeneralTypeIssues] https://github.com/rec/dtyper/issues/8eneralTypeIssues] https://github.com/rec/dtyper/issues/8
```
Comment (sujuka99) [inline kpops/cli/main.py:426]: typo

**Review by sujuka99 — CHANGES_REQUESTED** [review]

**Comment by sujuka99** [comment]
> should we remove those two options from the CLI now that they're global settings?

Sounds good

**Comment by raminqaf** [comment]
Since this is a breaking change (by removing the extra CLI parameters) please label it as `breaking-change`. One more thing: Please address the changes in the migration guide 😉

**Review by sujuka99 — APPROVED** [review]

File: docs/docs/user/migration-guide/v2-v3.md [hunk 3]
```diff
@@ -81,3 +81,16 @@ The `--config` flag in the CLI now points to the directory that contains `config
     enabled: true
     url: "http://my-custom-sr.url:8081"
 ```
+
+## [Define custom components module & pipeline base dir globally](https://github.com/bakdata/kpops/pull/387)
+
+The options for a custom `components_module` and `pipeline_base_dir` are now global settings, defined in `config.yaml`. The previous CLI parameters have been removed.
```
Comment (raminqaf) [inline docs/docs/user/migration-guide/v2-v3.md]: very tiny:
let's make the statement `The previous CLI parameters have been removed.` bold with a warning
https://squidfunk.github.io/mkdocs-material/reference/admonitions/#+type:warning
WDYT?

Comment (disrupted) [reply docs/docs/user/migration-guide/v2-v3.md]: good idea! I'll check to see if I did it correctly

**Review by raminqaf — APPROVED** [review]

**Pull request merged**